### ISSUE TITLE
BananaImmitation student brain set to internal error

### ISSUE THREAD

**Issue by emperor1412** [Unity-Technologies/ml-agents#503]
Hi, I set student brain to internal, then assign the model BananaI, remove the epsilon element in graph placeholder, then run and get this error:

```
UnityAgentsException: The tensorflow graph needs an input for state  of type DT_FLOAT
CoreBrainInternal.DecideAction (System.Collections.Generic.Dictionary`2[TKey,TValue] agentInfo) (at Assets/ML-Agents/Scripts/CoreBrainInternal.cs:334)
Brain.DecideAction () (at Assets/ML-Agents/Scripts/Brain.cs:241)
Academy.EnvironmentStep () (at Assets/ML-Agents/Scripts/Academy.cs:477)
Academy.FixedUpdate () (at Assets/ML-Agents/Scripts/Academy.cs:507)

```

**Comment by vincentpierre** [comment]
Hi,
We may have forgotten to replace the placeholder `state` with `vector_observation`. It says [here](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating-v0.3.md#unity-api) that you must `You must replace state with vector_observation and observation with visual_observation`. Can you try again with this change and report here if it worked ?
I hope this helps.

**Comment by emperor1412** [comment]
![image](https://user-images.githubusercontent.com/930373/37561786-27317412-2a8a-11e8-96df-1eca84c90f7c.png)

Hi, I did that, but still the same error

**Comment by vincentpierre** [comment]
You get the same error, with `state` in it still ? That is surprising. You have 2 brains in the imitation scene, make sure both of their internal brains have state replaced with vector_observation. Also, make sure you save the scene.

**Comment by emperor1412** [comment]
the other brain I have it in Player mode, not internal

**Comment by emperor1412** [comment]
hi this issue has been fixed in 72835e8188e4534ce2abdb961a6223d1637627bb
